libcontainer/cgroups/fs/cpuset.go

```diff
@@ -48,7 +50,109 @@ func (s *CpusetGroup) Remove(d *cgroupData) error {
 	return removePath(d.path("cpuset"))
 }
 
+func getNumericStat(path string, filename string, extracted *[]uint16) error {
+	fileContent, err := fscommon.GetCgroupParamString(path, filename)
+	if err != nil {
+		return err
+	}
+	if len(fileContent) == 0 {
+		return fmt.Errorf("%s found to be empty", filepath.Join(path, filename))
+	}
+
+	for _, s := range strings.Split(fileContent, ",") {
+		if strings.Contains(s, "-") {
+			splitted := strings.Split(s, "-")
+			if len(splitted) != 2 {
+				return fmt.Errorf("Couldn't parse %s", filepath.Join(path, filename))
+			}
+			min, err := strconv.ParseUint(splitted[0], 10, 16)
+			if err != nil {
+				return err
+			}
+			max, err := strconv.ParseUint(splitted[1], 10, 16)
+			if err != nil {
+				return err
+			}
+			if min > max {
+				return fmt.Errorf("Couldn't parse %s", filepath.Join(path, filename))
+			}
+			for i := min; i <= max; i++ {
+				*extracted = append(*extracted, uint16(i))
+			}
+		} else {
+			value, err := strconv.ParseUint(s, 10, 16)
+			if err != nil {
+				return err
+			}
+			*extracted = append(*extracted, uint16(value))
+		}
+
+	}
+
+	return nil
+}
+
 func (s *CpusetGroup) GetStats(path string, stats *cgroups.Stats) error {
+	var (
+		err  error
+		cpus []uint16
```

Comment:
You can avoid declaring it here if you modify `getNumericStat()` to return in. IOW
```go
cpus, err := getNumericStat(path, "cpuset.cpus")
```

Reply:
Applied